pkg/build/api/validation/validation.go

```diff
@@ -480,6 +471,15 @@ func validateTrigger(trigger *buildapi.BuildTriggerPolicy, fldPath *field.Path)
 			break
 		}
 		if trigger.ImageChange.From == nil {
+			if buildFrom == nil || buildFrom.Kind != "ImageStreamTag" {
+				invalidKindErr := field.Invalid(
+					fldPath.Child("imageChange"),
+					fmt.Sprintf("build from: %v", buildFrom),
+					"a default ImageChange trigger can only be used with builds based on ImageStreamTags.")
```

Comment:
how about "a default ImageChange trigger can only be used when the build strategy includes an ImageStreamTag reference"  ?